e2e/_suites/ingest-manager/features/fleet_mode_agent.feature

```diff
@@ -40,5 +36,33 @@ Scenario: Re-enrolling an agent
 @revoke-token
 Scenario: Revoking the enrollment token for an agent
   Given an agent is deployed to Fleet
+    And the agent is un-enrolled
+    And the "agent" process is "stopped" on the host
   When the enrollment token is revoked
-  Then an attempt to enroll a new agent fails
+  Then an attempt to enroll an agent with the old token fails
+
+@package-added-to-default-config
+Scenario: Adding packages to configurations
+  Given an agent is deployed to Fleet
+    And the package list API returns successfully
+  When the "Cisco" "latest" package version is installed successfully
+    And the "Cisco" package datasource is added to the "default" configuration
+  Then the "default" configuration shows the "Cisco" datasource added
+
+@new-agent-configuration
+Scenario: Assign an Agent to a new configuration
+  Given an agent is deployed to Fleet
+    And the agent is listed in Fleet as online
+  When a new configuration named "Test Fleet" is created
+    And the Agent is assigned to the configuration "Test Fleet"
+  Then a new enrollment token is created
+    And there is new data in the index from agent
+
+@new-configuration-new-package
+Scenario: Add a new config and a new package and assign an agent
+  Given an agent is deployed to Fleet
+  When a new configuration named "Test - custom logs" is created
+    And the "custom logs" package datasource is added to the "Test - custom logs" configuration
+    And the Agent is assigned to the configuration "Test - custom logs"
+  Then the "Test - custom logs" configuration shows the "custom logs" datasource added
+    And there is new data in the index from agent from "custom logs" stream
```

Comment:
I moved L67 from the `When` clause to the `Then` because after reading it, it makes me expect the datasource to be added, so it's seems a verification/assertion. Therefore, a `Then`